### PR TITLE
Use `SubTask.getOwnerExecutable`

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/jenkins/pull/7599 & https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/287.

### CODE REVIEW

**Comment by olamy** [comment]
the build is failing because of `RequireUpperBoundDeps` rule failures.

**Review by jglick — COMMENTED** [review]
@car-roll FYI

**Comment by jglick** [comment]
Checked locally with

```bash
mvnd clean install dependency:sources -Dtest=InjectedTest -DfailIfNoTests=false
```

so I hope this can finally go green before it gets more merge conflicts!

**Comment by olamy** [comment]
I merged it but since this commit ATH test fail in timeout.

**Comment by jglick** [comment]
Perhaps from the Jenkins core or BOM updates? https://ci.blueocean.io/job/blueocean/job/master/2181/execution/node/65/log/ shows lots of stuff indicating mistakes in test setup

```
WARNING: encountered /home/jenkins/workspace/blueocean_master/acceptance-tests/runner/runtime/target/jenkins1717351725658294775home/plugins/credentials.hpi under a nonstandard name; expected credentials.jpi
SEVERE: Failed Loading plugin Credentials Plugin v1254.vb_96f366e7b_a_d (credentials)
```

Anyway should you not turn off ci.blueocean.io by now? https://ci.blueocean.io/job/blueocean/job/master/ has not been stable since November.

**Comment by olamy** [comment]
> Perhaps from the Jenkins core or BOM updates? https://ci.blueocean.io/job/blueocean/job/master/2181/execution/node/65/log/ shows lots of stuff indicating mistakes in test setup
> 
> ```
> WARNING: encountered /home/jenkins/workspace/blueocean_master/acceptance-tests/runner/runtime/target/jenkins1717351725658294775home/plugins/credentials.hpi under a nonstandard name; expected credentials.jpi
> SEVERE: Failed Loading plugin Credentials Plugin v1254.vb_96f366e7b_a_d (credentials)
> ```
> 
> Anyway should you not turn off ci.blueocean.io by now? https://ci.blueocean.io/job/blueocean/job/master/ has not been stable since November.

yup there is an issue here https://github.com/jenkins-infra/helpdesk/issues/2954
I definitely agree it's not perfect (but not enough time to fix those) previously, the build was only unstable (e.g, known flaky ATH tests but now it hangs which ring some alarms to me)

File: pom.xml [hunk 1]
```diff
@@ -32,12 +32,11 @@
         when changing jenkins core version please remember to change it in Jenkinsfile as well jenkinsVersions
         and  acceptance-tests/runner/scripts/args.sh
     -->
-    <jenkins.version>2.361.4</jenkins.version>
```
Comment (basil) [inline pom.xml:35]: Implies a corresponding change in `jenkinsci/bom/bom-2.387.x/pom.xml`: https://github.com/jenkinsci/bom/pull/2395

**Pull request merged**